src/helpers/MailerHelper.php

```diff
@@ -82,10 +83,19 @@ public static function allMailerTransportTypes(): array
      */
     public static function createTransportAdapter(string $type, ?array $settings = null): TransportAdapterInterface
     {
-        return Component::createComponent([
+        $component = Component::createComponent([
             'type' => $type,
-            'settings' => $settings,
         ], TransportAdapterInterface::class);
+
+        if ($settings) {
+            if ($component instanceof Model) {
+                $component->setAttributes($settings);
```

Comment:
Should we be passing `false` to `$safeOnly` here?

Reply:
No – the whole point is to go through `setAttributes` _with_ `safeOnly` to avoid errors from stale PC values.